File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.merge(num, 1, Integer::sum);
        }

        List<Integer> list = map.entrySet()
                                 .stream()
                                 .sorted(Map.Entry.<Integer, Integer>comparingByValue().reversed())
                                 .limit(k)
                                 .map(Map.Entry::getKey)
                                 .collect(Collectors.toList());

        int[] arr = new int[k];

        for (int i = 0; i < k; i++) {
            arr[i] = list.get(i);
        }
        return arr;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        counts = {}

        for char in s:
            counts[char] = counts.get(char, 0) + 1

        for char in t:
            counts[char] = counts.get(char, 0) - 1

        for num in counts.values():
            if num !=0:
                return False
        return True
